db/view/view.cc

```diff
@@ -318,6 +320,50 @@ static bool is_partition_key_empty(
     }
 }
 
+// Checks if the result matches the provided view filter
+class view_filter_checking_visitor {
+    const schema& _base;
+    const view_info& _view;
+    const std::vector<bytes>& _pk;
+    const std::vector<bytes>& _ck;
+
+    bool _matches_view_filter = true;
+public:
+    view_filter_checking_visitor(const schema& base, const view_info& view, const std::vector<bytes>& pk, const std::vector<bytes>& ck)
+        : _base(base)
+        , _view(view)
+        , _pk(pk)
+        , _ck(ck)
+    {}
+
+    void accept_new_partition(const partition_key& key, uint64_t row_count) {}
+    void accept_new_partition(uint64_t row_count) {}
+
+    void accept_new_row(const clustering_key& key, const query::result_row_view& static_row, const query::result_row_view& row) {
+        _matches_view_filter = check_if_matches(static_row, row);
```

Comment:
Is it correct to overwrite _matches_view_filter here, or to combine with the existing value?  If there are multiple calls to accept_new_row, we'll only see the last row's result.

Reply:
Looking at how this visitor is used below, I believe accept_new_row will only be called once.  But this is fragile, and we can make it more robust by updating instead of overwriting _matches_view_filter, making this code more reusable tomorrow.

Reply:
Yes, the assumption is that we never check more than 1 row here. Still, updating instead of rewriting is trivial, so I'll do so